packages/lib/getBrandColours.tsx

```diff
@@ -0,0 +1,193 @@
+        return hex + hex;
+      })
+      .join("");
+  }
+
+  return hex;
+}
+
+export const createColorMap = (brandColor: string) => {
+  const response: Record<string, string> = {
+    500: `#${brandColor}`.replace("##", "#"),
+  };
+  const intensityMap: {
+    [key: number]: number;
+  } = {
+    50: 0.95,
+    100: 0.9,
+    200: 0.75,
+    300: 0.6,
+    400: 0.3,
+    600: 0.9,
+    700: 0.75,
+    800: 0.6,
+    900: 0.49,
+  };
+
+  [50, 100, 200, 300, 400].forEach((level) => {
+    response[level] = lighten(brandColor, intensityMap[level]);
+  });
+
+  [600, 700, 800, 900].forEach((level) => {
+    response[level] = darken(brandColor, intensityMap[level]);
+  });
+  return response;
+};
+
+function getWCAGContrastColor(background: string): string {
+  // Convert the hex background color to RGB
+  const { r, g, b } = hexToRgb(background);
+  // Calculate the luminance of the background color
+  const luminance = (0.2126 * r + 0.7152 * g + 0.0722 * b) / 255;
+
+  // If the luminance is less than 0.5, use white as the text color, otherwise use black
+  return luminance < 0.5 ? "#FFFFFF" : "#000000";
+}
+
+/**
+ * Given a light and dark brand color value, update the css variables
+ * within the document to reflect the new brand colors.
+ */
+const useGetBrandingColours = ({
+  lightVal = BRAND_COLOR,
+  darkVal = DARK_BRAND_COLOR,
+}: {
+  lightVal: string | undefined | null;
+  darkVal: string | undefined | null;
+}) => {
+  const embedBrandingColors = useBrandColors();
+
+  lightVal = embedBrandingColors.brandColor || lightVal;
+
+  // convert to 6 digit equivalent if 3 digit code is entered
+  lightVal = normalizeHexCode(lightVal, false);
+
+  darkVal = normalizeHexCode(darkVal, true);
+
+  const lightColourMap = createColorMap(lightVal);
+  const darkColourMap = createColorMap(darkVal);
+
+  const theme = {
+    light: {
+      "cal-brand": lightColourMap["500"],
+      "cal-brand-emphasis": lightColourMap["400"],
+      "cal-brand-subtle": lightColourMap["200"],
+      "cal-brand-text": getWCAGContrastColor(lightColourMap["500"]),
+    },
+    dark: {
+      "cal-brand": darkColourMap["500"],
+      "cal-brand-emphasis": darkColourMap["600"],
+      "cal-brand-subtle": darkColourMap["800"],
+      "cal-brand-text": getWCAGContrastColor(darkColourMap["500"]),
+    },
+    // @hariom - are these needed still? Can we move these to tokens?
+    root: {
+      "booking-highlight-color": embedBrandingColors.highlightColor || "#10B981",
+      "booking-lightest-color": embedBrandingColors.lightestColor || "#E1E1E1",
+      "booking-lighter-color": embedBrandingColors.lighterColor || "#ACACAC",
+      "booking-light-color": embedBrandingColors.lightColor || "#888888",
+      "booking-median-color": embedBrandingColors.medianColor || "#494949",
+      "booking-dark-color": embedBrandingColors.darkColor || "#313131",
+      "booking-darker-color": embedBrandingColors.darkerColor || "#292929",
+    },
+  };
+  return theme;
```

Comment:
@hariombalhara can we move these to normal tokens we use ? 

`bg-emphasis` `bg-subtle` etc? If not this PR might break embed styling - i tested and it seemed OKAY

Reply:
In my new booker I haven't implemented them yet, because I was waiting for these new variables to become available. I think it makes much more sense to replace them with the new variables, instead of keeping variables specific for only the booker. That way if users use these variables, and will use more of our atoms later, they all share the same variables and thus are styled right away.